.cirrus.star

```diff
@@ -1,4 +1,430 @@
+                "source cirrus-env QA",
+                "source .cirrus/use-gradle-wrapper.sh",
+                "./gradlew \"${GRADLE_TASK}\" \"-Dsonar.runtimeVersion=${SQ_VERSION}\" --info --build-cache --console plain --no-daemon"
+            ],
+            "cleanup_gradle_script": [
+                "/usr/bin/find \"${CIRRUS_WORKING_DIR}/.gradle/caches\" -type d -name \"8.*\" -prune -maxdepth 1 -exec rm -rf {} \\;",
+                "rm -rf \"${CIRRUS_WORKING_DIR}/.gradle/caches/journal-1/\""
+            ],
+        }
+    }
+
+
+def promote_task():
+    return {
+        "promote_task": {
+            "only_if": "$CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_TAG == \"\" && ($CIRRUS_PR != \"\" || $CIRRUS_BRANCH == $CIRRUS_DEFAULT_BRANCH || $CIRRUS_BRANCH =~ \"branch-.*\" || $CIRRUS_BRANCH =~ \"dogfood-on-.*\")",
+            "depends_on": [
+                "build",
+                "build_test_analyze",
+                "qa_os_win",
+                "qa_ruling",
+                "qa_plugin"
+            ],
+            "env": {
+                "ARTIFACTORY_PROMOTE_ACCESS_TOKEN": "VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-promoter access_token]",
+                "GITHUB_TOKEN": "VAULT[development/github/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-promotion token]",
+                "BURGR_URL": "VAULT[development/kv/data/burgr data.url]",
+                "BURGR_USERNAME": "VAULT[development/kv/data/burgr data.cirrus_username]",
+                "BURGR_PASSWORD": "VAULT[development/kv/data/burgr data.cirrus_password]",
+                "ARTIFACTS": "com.sonarsource.iac:sonar-iac-plugin:jar"
+            },
+            "eks_container": {
+                "image": "${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-latest",
+                "cluster_name": "${CIRRUS_CLUSTER_NAME}",
+                "region": "eu-central-1",
+                "namespace": "default",
+                "use_in_memory_disk": "true",
+                "cpu": 1,
+                "memory": "2G"
+            },
+            "project_version_cache": {
+                "folder": "$PROJECT_VERSION_CACHE_DIR",
+                "fingerprint_script": "echo $BUILD_NUMBER",
+                "populate_script": "mkdir -p \"$PROJECT_VERSION_CACHE_DIR\"\ntouch \"$PROJECT_VERSION_CACHE_DIR\"/evaluated_project_version.txt\n",
+                "reupload_on_changes": "true"
+            },
+            "script": [
+                "source cirrus-env PROMOTE",
+                "cirrus_jfrog_promote multi",
+                "source ${PROJECT_VERSION_CACHE_DIR}/evaluated_project_version.txt",
+                "github-notify-promotion",
+                "burgr-notify-promotion"
+            ]
+        }
+    }
+
+
 def main(ctx):
-    return load_features(ctx)
+    conf = dict()
+    merge_dict(conf, load_features(ctx))
+    merge_dict(conf, env())
+    merge_dict(conf, build_task())
```

Comment:
I like the fact that functions definitions are located in the file in the same order as here (and in the order as they are executed).